src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -389,15 +396,32 @@ class ContentOpenWith extends PureComponent<Props, State> {
                             integrations={integrations}
                         />
                     )}
-                    {shouldRenderIntegrationPortal && (
+                    {(shouldRenderLoadingIntegrationPortal ||
+                        shouldRenderErrorIntegrationPortal) && (
                         <IntegrationPortal
```

Comment:
Super minor, but you could consider moving this to a separate component to clean up this file's render method. The error/loading state seems to dominate it at the moment.